src/app/ReadClient.cpp

```diff
@@ -327,6 +331,7 @@ CHIP_ERROR ReadClient::OnUnsolicitedReportData(Messaging::ExchangeContext * apEx
 {
     mpExchangeCtx  = apExchangeContext;
     CHIP_ERROR err = ProcessReportData(std::move(aPayload));
+    mpExchangeCtx  = nullptr;
```

Comment:
@bzbarsky-apple If I understand correctly, `ProcessReportData` will trigger the transmission of a StatusResponse, which is the final message in this exchange, resulting in the closure of the exchange automatically.

So this setting of the `mpExchangeCtx` to nullptr is purely for cleanliness, and serves no functional value right?

Reply:
https://github.com/project-chip/connectedhomeip/issues/10074